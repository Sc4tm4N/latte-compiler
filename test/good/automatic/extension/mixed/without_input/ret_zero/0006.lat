
class A {
  int[] a;

  void change(int i, int x) {
    a[i] = x;
  }

  int get(int i) {
    return a[i];
  }
}

class B {
  int[] a;

  void change(int i, int x) {
    a[i] = x;
  }

  int get(int i) {
    return a[i];
  }
}

int main() {
  int[] a = new int[20];
  a[0]=0;a[1]=1;a[2]=2;a[3]=3;
  A ac = new A;
  ac.a = a;
  B bc = new B;
  bc.a = a;

  printInt(ac.a[0]);
  printInt(ac.a[1]);
  printInt(ac.a[2]);
  printInt(ac.a[3]);
  printInt(bc.a[0]);
  printInt(bc.a[1]);
  printInt(bc.a[2]);
  printInt(bc.a[3]);

  ac.change(10, 12);
  bc.change(15, 18);

  printInt(ac.a[10]);
  printInt(ac.a[15]);
  printInt(bc.a[10]);
  printInt(bc.a[15]);
  printInt(a[10]);
  printInt(a[15]);

  return 0;
}
