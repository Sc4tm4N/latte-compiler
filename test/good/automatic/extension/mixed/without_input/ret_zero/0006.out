0
1
2
3
0
1
2
3
12
18
12
18
12
18
